Update api-overview.md (#15896)
pingcap/docs: tidb-cloud/api-overview.md

- [Rate Limiting](https://docs.pingcap.com/tidbcloud/api/v1beta#section/Rate-Limiting)
- API Full References
    - [v1beta1](https://docs.pingcap.com/tidbcloud/api/v1beta1)
        - Scope: Billing 

>         - We will keep adding new APIs in v1beta1 version. 

Reviewer: 增加这些后，左侧的目录树会改变吗，会从 v1beta1 改成什么

Reviewer: @qqqdan

Author: 左侧的树暂时不会变。目前还是两个页面，v1beta1 和 v1beta 分别一个页面。 后续等 v1beta1 新加了内容，会有具体分类的link 加在树上

Reviewer: 那对用户来说，还是不能从左侧目录树找到 cluster mgmt 的入口的

>         - We will keep adding new APIs in v1beta1 version. 

Reviewer 2: ```suggestion
        - New APIs will continue to be added in the v1beta1 version. 
```

    - [v1beta](https://docs.pingcap.com/tidbcloud/api/v1beta#tag/Project)

>         - Scope: Dedicated Cluster management, Serverless Cluster management, Project management, Backup and Restore , Import. 
>         - Version v1beta has been finalized and no new API will be added in this version.

Reviewer 2: ```suggestion
        - Scope: TiDB Dedicated cluster management, TiDB Serverless cluster management, project management, backup and restore, import. 
        - The v1beta version has been finalized and no new APIs will be added in this version.
```
